commons/src/test/java/com/powsybl/commons/datasource/ZipArchiveDataSourceTest.java

```diff
@@ -0,0 +1,175 @@
+    private static final String BASENAME = "network";
+    private static final String MAIN_FILE = BASENAME + "." + MAIN_EXT;
+    private static final String ZIP_FILENAME = MAIN_FILE + ".zip";
+    private static final String ZIP_PATH = WORK_DIR + ZIP_FILENAME;
+    private static final String ADDITIONAL_SUFFIX = "_mapping";
+    private static final String ADDITIONAL_EXT = "csv";
+    private static final String ADDITIONAL_FILE = BASENAME + ADDITIONAL_SUFFIX + "." + ADDITIONAL_EXT;
+    private static final String UNRELATED_FILE = "other.de";
+
+    @Override
+    @BeforeEach
+    void setUp() throws Exception {
+        super.setUp();
+        archiveWithSubfolders = "foo.iidm.zip";
+        appendException = "append not supported in zip file data source";
+        archiveFormat = ArchiveFormat.ZIP;
+        compressionFormat = CompressionFormat.ZIP;
+    }
+
+    @Test
+    @Override
+    void testConstructors() {
+        // Observer
+        DataSourceObserver observer = new DefaultDataSourceObserver();
+
+        // Check constructors
+        checkDataSource(new ZipArchiveDataSource(testDir, "foo_bar.zip", "foo", observer), "foo_bar.zip", "foo", observer);
+        checkDataSource(new ZipArchiveDataSource(testDir, "foo_bar.zip", "foo"), "foo_bar.zip", "foo", null);
+        checkDataSource(new ZipArchiveDataSource(testDir, "foo", observer), "foo.zip", "foo", observer);
+        checkDataSource(new ZipArchiveDataSource(testDir, "foo"), "foo.zip", "foo", null);
+        checkDataSource(new ZipArchiveDataSource(testDir.resolve("foo_bar.zip")), "foo_bar.zip", "foo_bar", null);
+    }
+
+    private void checkDataSource(ZipArchiveDataSource dataSource, String zipFileName, String baseName, DataSourceObserver observer) {
+        assertEquals(testDir, dataSource.getDirectory());
+        assertEquals(zipFileName, dataSource.getArchiveFilePath().getFileName().toString());
+        assertEquals(baseName, dataSource.getBaseName());
+        assertEquals(observer, dataSource.getObserver());
+    }
+
+    @Override
+    protected boolean appendTest() {
+        return false;
+    }
+
+    @Override
+    protected DataSource createDataSource() {
+        return new ZipArchiveDataSource(testDir, "foo.zip", "foo");
+    }
+
+    @Override
+    protected DataSource createDataSource(DataSourceObserver observer) {
+        return new ZipArchiveDataSource(testDir, "foo", observer);
+    }
+
+    @Override
+    protected String getFileName(String baseName, String mainExtension, ArchiveFormat archiveFormat,
+                                 CompressionFormat compressionFormat) {
+        return testDir + "/" + baseName + (mainExtension.isEmpty() ? "" : ".") + mainExtension
+            + (compressionFormat == null ? "" : "." + compressionFormat.getExtension());
+    }
+
+    static Stream<Arguments> provideArgumentsForWriteThenReadTest() {
+        return Stream.of(
+            Arguments.of("foo", "iidm", CompressionFormat.ZIP),
```

Comment:
should use the same semantics for the "mainExtension" as other test classes

            Arguments.of("foo", "iidm", CompressionFormat.ZIP),
vs
            Arguments.of("foo", ".iidm", CompressionFormat.ZIP),

Reply:
(or just remove the mainExtension concept from this PR entirely, it's just used for tests and you can put everything in one string)

Reply:
done